Library/Formula/yap.rb

```diff
@@ -1,38 +1,84 @@
-require 'formula'
-
 class Yap < Formula
-  homepage 'http://www.dcc.fc.up.pt/~vsc/Yap/index.html'
-  url 'http://www.dcc.fc.up.pt/~vsc/Yap/yap-6.2.2.tar.gz'
-  sha1 'a02f80cac67c287645b2ced9502f5ea24a07f1c3'
+  homepage "https://www.dcc.fc.up.pt/~vsc/Yap/index.html"
+  url "https://www.dcc.fc.up.pt/~vsc/Yap/yap-6.2.2.tar.gz"
+  sha256 "f15f8382104443319a5883eafce5f52f4143b526c7f1cd88d19c1f63fc06d750"
 
   devel do
-    url 'http://www.dcc.fc.up.pt/~vsc/Yap/yap-6.3.3.tar.gz'
-    sha1 'd191e419e5cf74b11e003aae5fe148f3f2f26ac5'
+    url "https://www.dcc.fc.up.pt/~vsc/Yap/yap-6.3.3.tar.gz"
+    sha256 "aee3b449b1669af07a8291ce6e7fb0a9b35e1343e2ab96fadb1a37552397fa78"
   end
 
-  depends_on 'gmp'
-  depends_on 'readline'
+  depends_on "gmp"
+  depends_on "readline"
+  depends_on "homebrew/versions/gcc49"
+  depends_on :java => "1.8+"
 
   fails_with :clang do
     cause "uses variable-length arrays in structs, which will never be supported by clang"
   end
 
+  stable do
+    # http://permalink.gmane.org/gmane.comp.ai.prolog.yap.general/1790
+    patch :DATA
```

Comment:
Yes, I’m still experimenting here, this PR is not merge-ready (and it fails on devel).